agent/lib/src/api/model.rs

```diff
@@ -0,0 +1,187 @@
+use serde::{Deserialize, Serialize};
+
+
+/// Represents a file to be included in the workspace
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    /// Name of the file, paths relative to the workspace
+    pub filename: String,
+    /// Content of the file
+    pub content: String,
+}
+
+/// Identifies the type of the message
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    /// Status message to indicate that the agent is ready
+    Status,
+    /// Request message
+    Request,
+    /// Response message answering to a request message
+    Response,
+}
+
+/// Code to tell what the Request/Response message is about
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Code {
+    /// Represents a request to run the code or a response to such request
+    Run,
+    /// Indicates that your status is that you are ok and ready to communicate
```

Comment:
```suggestion
    /// Agent is ready to communicate.
```

Please rename it to `Ready`at least.